math/src/elliptic_curve/short_weierstrass/curves/bls12_381/curve.rs

```diff
@@ -1,13 +1,25 @@
 use super::field_extension::{BLS12381PrimeField, Degree2ExtensionField};
+use crate::cyclic_group::IsGroup;
 use crate::elliptic_curve::short_weierstrass::point::ShortWeierstrassProjectivePoint;
 use crate::elliptic_curve::traits::IsEllipticCurve;
+use crate::unsigned_integer::element::*;
 use crate::{
     elliptic_curve::short_weierstrass::traits::IsShortWeierstrass, field::element::FieldElement,
 };
 
 pub type BLS12381FieldElement = FieldElement<BLS12381PrimeField>;
 pub type BLS12381TwistCurveFieldElement = FieldElement<Degree2ExtensionField>;
 
+/// Cube Root of Unity in G1.
+pub const CUBE_ROOT_OF_UNITY_G1: U384 = U384::from_hex_unchecked(
+    "5f19672fdf76ce51ba69c6076a0f77eaddb3a93be6f89688de17d813620a00022e01fffffffefffe",
```

Comment:
Make this of type `FieldElement`. This will put it in Montgomery form once at compile time, instead of doing it at every call of `phi`.
```suggestion
pub const CUBE_ROOT_OF_UNITY_G1: FieldElement<BLS12381PrimeField> = FieldElement::from_hex_unchecked(
    "5f19672fdf76ce51ba69c6076a0f77eaddb3a93be6f89688de17d813620a00022e01fffffffefffe",
);
```